### ISSUE TITLE
syntax-check[specific]: A template was resolved to an Omit scalar.

### ISSUE THREAD

**Issue by felix-grzelka** [ansible/ansible-lint#5135]
##### Summary

Ansible lint raises an error, when `vars_file` is evaluated to omit.
It seems its not possible to disable this error on just the line it is caused by.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

The File `ansible_vault_secrets.yml` does not exist, so the template gets evaluated to `omit`.
It works when running the playbook, but ansible-lint complains, when the file does not exist.

```console (paste below)
--- 
- name: "Something with vaultvars"
  hosts: "all"
  gather_facts: false
  vars_files:
    - "{{ 'ansible_vault_secret.yml' is exists | ternary('ansible_vault_secrets.yml', omit) }}"
  tasks:   
    - name: "Hello World"
      debug:
        msg: "Hello World"
```

##### Desired Behavior

Ansible lint finds no error, or the error can be disabled on just one line.

Possible security bugs should be reported via email to `security@ansible.com`

##### Actual Behavior

Please give some details of what is happening. Include a [minimum complete
verifiable example] with:

- minimized playbook to reproduce the error
- the output of running ansible-lint including the command line used
- if you're getting a stack trace, also the output of
  `ansible-playbook --syntax-check playbook`

```paste below
syntax-check[specific]: A template was resolved to an Omit scalar.
```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by Jkhall81** [comment]
Hi @felix-grzelka

This is how omit is supposed to work.  This is from Ansible's core parser, and not ansible-lint itself.

The omit variable is just for task module parameters.  It cannot omit items inside YAML lists or top-level keys like vars_files.  When the file is missing, the template returns Ansible's internal __omit_place_holder__ object, and then ansible-lint will fail with a syntax error.

I'm not sure if this fits your use case.  but:

```yaml
- name: "Something with vaultvars"
  hosts: "all"
  gather_facts: false
  tasks:
    - name: Load vault secrets if file exists
      ansible.builtin.include_vars:
        file: ansible_vault_secrets.yml
      when: "'ansible_vault_secrets.yml' is file"

    - name: "Hello World"
      ansible.builtin.debug:
        msg: "Hello World"
```

This will use ansible_vault_secrets.yml if it exists, and if not it'll skip this task.
